task9/task9.js

```diff
@@ -0,0 +1,512 @@
+var adList = [
+    {
+        id: '1',
+            description: 'Скидка на стулья - до 15%',
+            createdAt: new Date('2021-03-15T23:00:00'),
+            link: 'https://coolchairs.com',
+            vendor: 'Chair Service',
+            photoLink: 'https://images.app.goo.gl/dgAFyP2cEduzkJUP9',
+            validUntil: new Date('2021-06-01T23:00:00'),
+            discount: '15%',
+            hashTags: ['tag1', 'tag2']
```

Comment:
Missing comma, which breaks the code completely, so the answer to previous question is negative, I guess.

Reply:
Okay, I'll take that into account and correct the mistakes, but in the end, what grade will I get for the 9th task?

Reply:
The biggest mistake of yours in this task is the fact that this code has been passed to review 3 days before you by other guy, and wasn't even tested 🙁  
So the score is:
- -2 for deadline (you sent this task to review 3 days ago, while deadline was on April 19)
- -3 for not working functionality. Even though I wasn't able to run your code, I already see vendor select items are hardcoded, there's no way to log in/out (in original code this functionality is missing too). Also that HTML array sorting (why to do it?) and the fact that script is not included to any HTML file, which is strange, since I can imagine that you could test task7, for example, without using HTML file, but not this task. Also all this point is missing:

> 3. Реализовать функции для демонстрации работы бизнес-логики приложения
Требуется в глобальной области видимости реализовать функции, которые продемонстрируют работу бизнес-логики приложения, манипулируя DOMом. 
    Например, метод addAd/addPost(ad: object), должен вызвать соответствующий метод из модуля для работы с моделью данных, и, если объявление было добавлено успешно, вызвать метод для отображения изменений в DOMе (т.е. вызвать метод, разработанный в рамках пункта 1).

- -2 for originality of code 🤷‍♀️ 

So grade is 10 - 2 - 3 - 2 = 3.

Reply:
хорошо,можно тогда я задам последний вопрос,я прсото посчитал и мне чуть-чуть не хватает балла,можно ли сегодня еще скинуть 10 задачу?